src/components/button/button.component.tsx

```diff
@@ -131,12 +131,14 @@ RenderChildrenProps) {
     bgSize: "extra-small",
   };
 
+  const isValidChildren = children !== undefined && children !== false;
+
   return (
     <>
-      {iconType && iconPosition === "before" && children && (
+      {iconType && iconPosition === "before" && isValidChildren && (
```

Comment:
suggestion: we could just use Boolean(children) to achieve the same goal

Reply:
not really, because the `Boolean` will convert to `false` a falsy value resulting in the same behavior as before. e.g.
```Boolean(0) // false```

Reply:
Yeah fair point and forgot the point of the fix. No worries!